californium-core/src/main/java/org/eclipse/californium/core/network/stack/BlockwiseStatus.java

```diff
@@ -231,6 +235,35 @@ final synchronized int getBlockCount() {
 		return body;
 	}
 
+	/**
+	 * Get the endpoint-context to be used for followup block requests.
+	 * 
+	 * Use the endpoint-context of the response to support notifies from
+	 * different addresses. Restores the
+	 * {@link DtlsEndpointContext#KEY_HANDSHAKE_MODE}, if the value is
+	 * {@link DtlsEndpointContext#HANDSHAKE_MODE_NONE}.
+	 * 
+	 * @param blockContext endpoint-context to be used/adapted for follow-up
+	 *            requests.
+	 * @return endpoint-context for follow-up-requests
+	 */
+	synchronized EndpointContext getFollowUpEndpointContext(EndpointContext blockContext) {
+		if (followUpEndpointContext == null
+				|| !followUpEndpointContext.getPeerAddress().equals(blockContext.getPeerAddress())) {
+			// considering notifies with address changes,
+			// use the response's endpoint-context to compensate that
+			EndpointContext context = exchange.getRequest().getDestinationContext();
+			String mode = context.get(DtlsEndpointContext.KEY_HANDSHAKE_MODE);
+			if (mode != null && mode.equals(DtlsEndpointContext.HANDSHAKE_MODE_NONE)) {
+				// restore handshake-mode "none"
+				blockContext = MapBasedEndpointContext.addEntries(blockContext, DtlsEndpointContext.KEY_HANDSHAKE_MODE,
+						DtlsEndpointContext.HANDSHAKE_MODE_NONE);
+			}
+			followUpEndpointContext = blockContext;
```

Comment:
clearly a details but why not ? 

```java
if (mode != null && mode.equals(DtlsEndpointContext.HANDSHAKE_MODE_NONE)) {
	// restore handshake-mode "none"
	followUpEndpointContext = MapBasedEndpointContext.addEntries(blockContext, DtlsEndpointContext.KEY_HANDSHAKE_MODE, DtlsEndpointContext.HANDSHAKE_MODE_NONE);
} else {
	followUpEndpointContext = blockContext;
}
```

Reply:
If you prefer ...

Reply:
Don't do it if you don't think, it's clearer.

Reply:
Done ... over the time I get used to the other style, mainly to initialize "final" stuff. But I have no real preference.